diff --git a/src/CMake/nativeTests.cmake b/src/CMake/nativeTests.cmake
index ed6b1131f..84343f26f 100644
--- a/xrt/XRT/src/CMake/nativeTests.cmake
+++ b/xrt/XRT/src/CMake/nativeTests.cmake
@@ -18,14 +18,19 @@ add_test(NAME xrt-smi
   COMMAND ${XRT_BINARY_DIR}/runtime_src/core/tools/xbutil2/xrt-smi examine
   WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})
 
+if (NOT XRT_UPSTREAM)
 set_tests_properties(xrt-smi PROPERTIES ENVIRONMENT
   "XILINX_XRT=${XRT_BUILD_INSTALL_DIR}")
+endif()
 
 if (XRT_XRT OR XRT_ALVEO)
   add_test(NAME xbmgmt2
     COMMAND ${XRT_BINARY_DIR}/runtime_src/core/tools/xbmgmt2/xbmgmt examine -r host
     WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})
 
+if (NOT XRT_UPSTREAM)
   set_tests_properties(xbmgmt2 PROPERTIES ENVIRONMENT
     "XILINX_XRT=${XRT_BUILD_INSTALL_DIR}")
 endif()
+
+endif()
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index b5b0720..227353b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -7,5 +7,6 @@ set(XRT_BUILD_NUMBER 75)
 set(XRT_UPSTREAM_DEBIAN 1)
 set(XRT_UPSTREAM 1)
 
+enable_testing()
 add_subdirectory(xrt)
 add_subdirectory(xdna)
