diff --git a/src/runtime_src/core/include/xrt/deprecated/xclerr.h b/src/runtime_src/core/include/xrt/deprecated/xclerr.h
index f660a3624..5a519684a 100644
--- a/xrt/XRT/src/runtime_src/core/include/xrt/deprecated/xclerr.h
+++ b/xrt/XRT/src/runtime_src/core/include/xrt/deprecated/xclerr.h
@@ -1,34 +1,8 @@
-/*
- * Copyright (C) 2017 Xilinx, Inc
- * Author: Umang Parekh
- *
- * This file is dual licensed.  It may be redistributed and/or modified
- * under the terms of the Apache 2.0 License OR version 2 of the GNU
- * General Public License.
- *
- * Apache License Verbiage
- *
- * Licensed under the Apache License, Version 2.0 (the "License"). You may
- * not use this file except in compliance with the License. A copy of the
- * License is located at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
- * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
- * License for the specific language governing permissions and limitations
- * under the License.
- *
- * GPL license Verbiage:
- * This program is free software; you can redistribute it and/or modify
- * it under the terms of the GNU General Public License as published by the Free Software Foundation;
- * either version 2 of the License, or (at your option) any later version.
- * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
- * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
- * See the GNU General Public License for more details.
- * You should have received a copy of the GNU General Public License along with this program;
- * if not, write to the Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+/**
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-License-Identifier: GPL-2.0
+ * Copyright (C) 2017 Xilinx, Inc. All rights reserved.
+ * Copyright (C) 2026 Advanced Micro Devices, Inc. All rights reserved.
  */
 
 /**
diff --git a/src/runtime_src/core/include/xrt/detail/ert.h b/src/runtime_src/core/include/xrt/detail/ert.h
index 46fabd934..10f4f06c5 100644
--- a/xrt/XRT/src/runtime_src/core/include/xrt/detail/ert.h
+++ b/xrt/XRT/src/runtime_src/core/include/xrt/detail/ert.h
@@ -1,38 +1,8 @@
-/*
- *  Copyright (C) 2019-2022, Xilinx Inc
- *
- *  This file is dual licensed.  It may be redistributed and/or modified
- *  under the terms of the Apache 2.0 License OR version 2 of the GNU
- *  General Public License.
- *
- *  Apache License Verbiage
- *
- *  Licensed under the Apache License, Version 2.0 (the "License");
- *  you may not use this file except in compliance with the License.
- *  You may obtain a copy of the License at
- *
- *  http://www.apache.org/licenses/LICENSE-2.0
- *
- *  Unless required by applicable law or agreed to in writing, software
- *  distributed under the License is distributed on an "AS IS" BASIS,
- *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- *  See the License for the specific language governing permissions and
- *  limitations under the License.
- *
- *  GPL license Verbiage:
- *
- *  This program is free software; you can redistribute it and/or
- *  modify it under the terms of the GNU General Public License as
- *  published by the Free Software Foundation; either version 2 of the
- *  License, or (at your option) any later version.  This program is
- *  distributed in the hope that it will be useful, but WITHOUT ANY
- *  WARRANTY; without even the implied warranty of MERCHANTABILITY or
- *  FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public
- *  License for more details.  You should have received a copy of the
- *  GNU General Public License along with this program; if not, write
- *  to the Free Software Foundation, Inc., 59 Temple Place, Suite 330,
- *  Boston, MA 02111-1307 USA
- *
+/**
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-License-Identifier: GPL-2.0
+ * Copyright (C) 2019-2022 Xilinx, Inc. All rights reserved.
+ * Copyright (C) 2022-2026 Advanced Micro Devices, Inc. All rights reserved.
  */
 
 /**
diff --git a/src/runtime_src/core/include/xrt/detail/xclbin.h b/src/runtime_src/core/include/xrt/detail/xclbin.h
index bf7d2ac02..97e8771d8 100644
--- a/xrt/XRT/src/runtime_src/core/include/xrt/detail/xclbin.h
+++ b/xrt/XRT/src/runtime_src/core/include/xrt/detail/xclbin.h
@@ -1,39 +1,8 @@
 /**
- *  Copyright (C) 2015-2024, Advanced Micro Devices, Inc. All rights Reserved.
- *
- *  This file is dual licensed.  It may be redistributed and/or modified
- *  under the terms of the Apache 2.0 License OR version 2 of the GNU
- *  General Public License.
- *
- *  Apache License Verbiage
- *
- *  Licensed under the Apache License, Version 2.0 (the "License");
- *  you may not use this file except in compliance with the License.
- *  You may obtain a copy of the License at
- *
- *  http://www.apache.org/licenses/LICENSE-2.0
- *
- *  Unless required by applicable law or agreed to in writing, software
- *  distributed under the License is distributed on an "AS IS" BASIS,
- *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- *  See the License for the specific language governing permissions and
- *  limitations under the License.
- *
- *  GPL license Verbiage:
- *
- *  This program is free software; you can redistribute it and/or modify
- *  it under the terms of the GNU General Public License as published by
- *  the Free Software Foundation, either version 2 of the License, or
- *  (at your option) any later version.
- *
- *  This program is distributed in the hope that it will be useful,
- *  but WITHOUT ANY WARRANTY; without even the implied warranty of
- *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- *  GNU General Public License for more details.
- *
- *  You should have received a copy of the GNU General Public License
- *  along with this program.  If not, write to the Free Software Foundation,
- *  Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-License-Identifier: GPL-2.0
+ * Copyright (C) 2015-2022 Xilinx, Inc. All rights reserved.
+ * Copyright (C) 2022-2026 Advanced Micro Devices, Inc. All rights reserved.
  */
 #ifndef INCLUDE_XRT_DETAIL_XCLBIN_H_
 #define INCLUDE_XRT_DETAIL_XCLBIN_H_
diff --git a/src/runtime_src/core/include/xrt/detail/xrt_error_code.h b/src/runtime_src/core/include/xrt/detail/xrt_error_code.h
index 63d961b92..65e4a2a77 100644
--- a/xrt/XRT/src/runtime_src/core/include/xrt/detail/xrt_error_code.h
+++ b/xrt/XRT/src/runtime_src/core/include/xrt/detail/xrt_error_code.h
@@ -1,38 +1,8 @@
-/*
- *  Copyright (C) 2020, Xilinx Inc
- *
- *  This file is dual licensed.  It may be redistributed and/or modified
- *  under the terms of the Apache 2.0 License OR version 2 of the GNU
- *  General Public License.
- *
- *  Apache License Verbiage
- *
- *  Licensed under the Apache License, Version 2.0 (the "License");
- *  you may not use this file except in compliance with the License.
- *  You may obtain a copy of the License at
- *
- *  http://www.apache.org/licenses/LICENSE-2.0
- *
- *  Unless required by applicable law or agreed to in writing, software
- *  distributed under the License is distributed on an "AS IS" BASIS,
- *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- *  See the License for the specific language governing permissions and
- *  limitations under the License.
- *
- *  GPL license Verbiage:
- *
- *  This program is free software; you can redistribute it and/or
- *  modify it under the terms of the GNU General Public License as
- *  published by the Free Software Foundation; either version 2 of the
- *  License, or (at your option) any later version.  This program is
- *  distributed in the hope that it will be useful, but WITHOUT ANY
- *  WARRANTY; without even the implied warranty of MERCHANTABILITY or
- *  FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public
- *  License for more details.  You should have received a copy of the
- *  GNU General Public License along with this program; if not, write
- *  to the Free Software Foundation, Inc., 59 Temple Place, Suite 330,
- *  Boston, MA 02111-1307 USA
- *
+/**
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-License-Identifier: GPL-2.0
+ * Copyright (C) 2020-2022 Xilinx, Inc. All rights reserved.
+ * Copyright (C) 2026 Advanced Micro Devices, Inc. All rights reserved.
  */
 #ifndef INCLUDE_XRT_DETAIL_XRT_ERROR_CODE_H_
 #define INCLUDE_XRT_DETAIL_XRT_ERROR_CODE_H_
diff --git a/src/runtime_src/core/include/xrt/detail/xrt_mem.h b/src/runtime_src/core/include/xrt/detail/xrt_mem.h
index 1e8001792..957d47562 100644
--- a/xrt/XRT/src/runtime_src/core/include/xrt/detail/xrt_mem.h
+++ b/xrt/XRT/src/runtime_src/core/include/xrt/detail/xrt_mem.h
@@ -1,30 +1,9 @@
-/*
- * Copyright (C) 2019-2022, Xilinx Inc - All rights reserved.
- * Xilinx Runtime (XRT) APIs
- *
- * Licensed under the Apache License, Version 2.0 (the "License"). You may
- * not use this file except in compliance with the License. A copy of the
- * License is located at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
- * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
- * License for the specific language governing permissions and limitations
- * under the License.
- *
- * GPL license Verbiage:
- * This program is free software; you can redistribute it and/or modify
- * it under the terms of the GNU General Public License as published by the Free Software Foundation;
- * either version 2 of the License, or (at your option) any later version.
- * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
- * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
- * See the GNU General Public License for more details.
- * You should have received a copy of the GNU General Public License along with this program;
- * if not, write to the Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+/**
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-License-Identifier: GPL-2.0
+ * Copyright (C) 2019-2022 Xilinx, Inc. All rights reserved.
+ * Copyright (C) 2022-2026 Advanced Micro Devices, Inc. All rights reserved.
  */
-
 #ifndef INCLUDE_XRT_DETAIL_XRT_MEM_H_
 #define INCLUDE_XRT_DETAIL_XRT_MEM_H_
 
