Submodule src/xrt/XRT contains modified content
diff --git a/src/xrt/XRT/src/runtime_src/hip/api/hip_memory.cpp b/src/xrt/XRT/src/runtime_src/hip/api/hip_memory.cpp
index 06e191565..f157910fd 100644
--- a/xrt/XRT/src/runtime_src/hip/api/hip_memory.cpp
+++ b/xrt/XRT/src/runtime_src/hip/api/hip_memory.cpp
@@ -30,7 +30,7 @@ namespace xrt::core::hip
     auto hip_mem = std::make_shared<xrt::core::hip::memory>(dev, size);
     auto address = hip_mem->get_address();
     throw_if(!address, hipErrorOutOfMemory, "Error allocating memory using hipMalloc!");
-      
+
     memory_database::instance().insert(reinterpret_cast<uint64_t>(address), size, std::move(hip_mem));
     *ptr = reinterpret_cast<void* >(address);
   }
@@ -49,11 +49,11 @@ namespace xrt::core::hip
     auto hip_mem = std::make_shared<xrt::core::hip::memory>(dev, size, flags);
     auto address = hip_mem->get_address();
     throw_if(!address, hipErrorOutOfMemory, "Error allocating memory using hipHostMalloc!");
-      
+
     memory_database::instance().insert(reinterpret_cast<uint64_t>(address), size, std::move(hip_mem));
     *ptr = address;
   }
-  
+
   // Register host memory so it can be accessed from the current device.
   static void
   hip_host_register(void* host_ptr, size_t size, unsigned int flags)
@@ -70,7 +70,7 @@ namespace xrt::core::hip
 
     memory_database::instance().insert(reinterpret_cast<uint64_t>(host_addr), size, std::move(hip_mem));
   }
-  
+
   // Get Device pointer from Host Pointer allocated through hipHostMalloc().
   static void
   hip_host_get_device_pointer(void** device_ptr, void* host_ptr, unsigned int flags)
@@ -105,7 +105,7 @@ namespace xrt::core::hip
 
     memory_database::instance().remove(reinterpret_cast<uint64_t>(ptr));
   }
-  
+
   // Free memory allocated by the hipHostMalloc().
   static void
   hip_host_free(void* ptr)
@@ -247,7 +247,7 @@ namespace xrt::core::hip
   }
 
   static void
-  hip_memcpy_host2device_async(hipDeviceptr_t dst, void* src, size_t size, hipStream_t stream)
+  hip_memcpy_host2device_async(hipDeviceptr_t dst, const void* src, size_t size, hipStream_t stream)
   {
     throw_invalid_value_if(!src, "src is nullptr.");
 
@@ -259,7 +259,7 @@ namespace xrt::core::hip
 
     auto hip_stream = get_stream(stream);
     throw_invalid_value_if(!hip_stream, "Invalid stream handle.");
-   
+
     // ptr to a xrt::core::hip::command object could be shared between global command_cache and stream::m_top_event::m_chain_of_commands of a stream object
     auto s_hdl = hip_stream.get();
     auto cmd_hdl = insert_in_map(command_cache,
@@ -413,7 +413,7 @@ hipMemset(void* dst, int value, size_t size)
 }
 
 hipError_t
-hipMemcpyHtoDAsync(hipDeviceptr_t dst, void* src, size_t size, hipStream_t stream)
+hipMemcpyHtoDAsync(hipDeviceptr_t dst, const void* src, size_t size, hipStream_t stream)
 {
   return handle_hip_func_error(__func__, hipErrorRuntimeMemory, [&] {
     xrt::core::hip::hip_memcpy_host2device_async(dst, src, size, stream);
@@ -463,4 +463,4 @@ hipMemPrefetchAsync(const void* dev_ptr, size_t count, int device, hipStream_t s
   return handle_hip_func_error(__func__, hipErrorRuntimeMemory, [&] {
     xrt::core::hip::hip_mem_prefetch_async(dev_ptr, count, device, stream);
   });
-}
\ No newline at end of file
+}
diff --git a/src/xrt/XRT/src/runtime_src/hip/api/hip_module.cpp b/src/xrt/XRT/src/runtime_src/hip/api/hip_module.cpp
index 12dcdd3e1..5c22bc4d7 100644
--- a/xrt/XRT/src/runtime_src/hip/api/hip_module.cpp
+++ b/xrt/XRT/src/runtime_src/hip/api/hip_module.cpp
@@ -20,7 +20,6 @@ hip_module_launch_kernel(hipFunction_t f, uint32_t /*gridDimX*/, uint32_t /*grid
                          void** kernelParams, void** extra)
 {
   throw_invalid_resource_if(!f, "function is nullptr");
-  throw_invalid_value_if(!kernelParams, "kernel parameters is nullptr");
 
   auto func_hdl = reinterpret_cast<function_handle>(f);
   auto hip_mod = module_cache.get(static_cast<function*>(func_hdl)->get_module());
@@ -277,7 +276,7 @@ hipModuleLoad(hipModule_t* module, const char* fname)
     // Treat fname passed is filepath to full ELF and
     // try creating full ELF module
     // if it throws fallback to xclbin + ELF flow
-    xrt::core::hip::module_handle handle;
+    xrt::core::hip::module_handle handle = nullptr;
     if (xrt::core::hip::hip_module_file_is_elf(fname)) {
       handle = xrt::core::hip::create_full_elf_module(std::string{fname});
       *module = reinterpret_cast<hipModule_t>(handle);
@@ -304,4 +303,3 @@ hipFuncSetAttribute(const void* func, hipFuncAttribute attr, int value)
     xrt::core::hip::hip_func_set_attribute(func, attr, value);
   });
 }
-
diff --git a/src/xrt/XRT/src/runtime_src/hip/core/memory.cpp b/src/xrt/XRT/src/runtime_src/hip/core/memory.cpp
index 135727348..4a86646f3 100644
--- a/xrt/XRT/src/runtime_src/hip/core/memory.cpp
+++ b/xrt/XRT/src/runtime_src/hip/core/memory.cpp
@@ -40,7 +40,7 @@ namespace xrt::core::hip
   }
 
   memory::memory(device* dev, size_t sz, unsigned int flags)
-      : m_device(std::move(dev)),
+      : m_device(dev),
 	m_size(sz),
 	m_type(memory_type::host),
 	m_flags(flags)
