diff --git a/src/xrt/XRT/src/runtime_src/core/tools/CMakeLists.txt b/src/xrt/XRT/src/runtime_src/core/tools/CMakeLists.txt
index 0d305d7c6..14b99155b 100644
--- a/xrt/XRT/src/runtime_src/core/tools/CMakeLists.txt
+++ b/xrt/XRT/src/runtime_src/core/tools/CMakeLists.txt
@@ -6,10 +6,11 @@ if (${XRT_BOOST_VERSION} VERSION_LESS 1.64.0)
 endif()
 
 xrt_add_subdirectory(xbutil2)
-if(CMAKE_VERSION VERSION_LESS "3.18.0")
+
+if(NOT XRT_ENABLE_XBTRACER OR CMAKE_VERSION VERSION_LESS "3.18.0")
   # CPackDeb.cmake does not support -l{dir} which is required
   # when packaging plugin library libxrt_capture.so 
-  message(WARNING "CMake version is less than 3.18.0, build of xbtracer disabled")
+  message(WARNING "Build of xbtracer disabled")
 else()
   xrt_add_subdirectory(xbtracer)
 endif()
